### ISSUE TITLE
Transition of 2 pictures : transparence

### ISSUE THREAD

**Issue by wolfnco** [OpenShot/openshot-qt#1837]
Hello,

I use OpenShot to make a slideshow of picture for a wedding.

I have an issue with transitions. It's probably nothing, but I don't find the solution.
When the picture changes from landscape to portrait, at the end of the transition, the first picture is still visible instead of the black blocks at left and right of the second picture, until the end of the juxtaposition.
How to make these black blocs non-transparent?

Below a screenshot of the issue.

Thank you !

**System Details**
 - Operating System / Distro: Windows 10 pro
 - OpenShot Version: 2.4.2

**Screenshots**

![image](https://user-images.githubusercontent.com/36747565/42599145-9d016358-855e-11e8-979a-586b1a896324.png)

**Comment by peanutbutterandcrackers** [comment]
Hmm... I have run into this issue myself, too. I wonder if @N3WWN has some idea as to how this can be achieved.

In the meantime, you could read [this guide](https://gist.github.com/peanutbutterandcrackers/f0f666243133e0ed25abbc12a4ba23d7) to learn all about OpenShot. You could use keyframes to have one image move away and another one comes in. :+1:

**Comment by wolfnco** [comment]
Hello peanutbutterand
Thank you for your answer.

I have made few tests. The issue is not related to transition. If i juxtapose 2 pictures in the timeline (first landscape, 2nd portrait), The first picture is visible instead of the black bars of the second picture.

So I guess Openshot add transparency when it add  black bars to wrong format picture. But how to remove it?

I've take a look to keyframes but i have thousands of picture to show... so i need to automatically add the slideshow.

**Comment by wolfnco** [comment]
As a workaround, i use fade in/fade out instead of transition

**Comment by N3WWN** [comment]
@wolfnco - Fading is what I use when I want the same output effect that you're looking for.  I alternate which track the next clip goes onto, overlap them by the "fast fade" amount (0.5s?) and apply `Fade`->`Entire Clip`->`Fade In and Out (Fast)` to all of the items.

**Comment by wolfnco** [comment]
I noticed something strange yesterday :

I opened a previous project, size about 0.5MB.
I added a slideshow (60 pictures), all worked fine.
Few minutes later, when  I tried to add a new object to the timeline, Openshot became very very slow (few minutes for each modification!). I'm using Win10 64 bits 16GB.

The project file size was about 70MB at this moment.

A long time later, Openshot was again fast, and the project file size decreased to 0.6MB !

I'm curious. How does Openshot manage its project file? Does it use it as a workplace, and sometimes it purges it?

Thank you

**Comment by peanutbutterandcrackers** [comment]
Wow. I am not aware of such behaviour. Perhaps @N3WWN or @DylanC could shed some light into it? It is definitely strange... :thinking:

**Comment by wolfnco** [comment]
I have a old backup of this project, when I had the same issue.  51MB. It seems to be very huge for text only!

Enclosed the compressed file, if you want to take a look.

[montage.zip](https://github.com/OpenShot/openshot-qt/files/2191950/montage.zip)

**Comment by peanutbutterandcrackers** [comment]
Whoa! That is a huge file. And my text editior's getting frozen while trying to read it... Strange, it compressed to 1.7 MB. Cool.

I think @N3WWN is better at doing such detective works, TBH.

Thank you for posting the file though. It will be useful in making OpenShot better. :)

**Comment by wolfnco** [comment]
Text files are compressed well :) I was able to open it with notepad++ (after 30secondes of freeze)

Maybe it may explain the performance issue related by many win10 64bits users!

**Comment by N3WWN** [comment]
Looks like a bug in the undo/redo history.

The file is almost entirely undo history.  

It looks like the undo history is captured as part of each new undo history that is captured, so there is a snowball effect until the old history starts dropping off, at which point the history starts shrinking again.

- First undo contains no history, just the current change.
- Second undo contains the first undo history and the current change.
- Third undo contains the second undo history (which also contains the first undo history), the first undo history and the current change.
- Fourth undo contains the third undo history (which also contains the second undo history which also contains the first undo history), the second undo history (which also contains the first undo history), the first undo history and the current change.
- etc

Looking at the .osp file, there are : 
- 95141 nodes under Root->History->undo (20)->Item[0]
- 95141 nodes under Root->History->undo (20)->Item[1]
- 95141 nodes under Root->History->undo (20)->Item[2]
- 95141 nodes under Root->History->undo (20)->Item[3]
- 95141 nodes under Root->History->undo (20)->Item[4]
- 95141 nodes under Root->History->undo (20)->Item[5]
- 20 nodes under Root->History->undo (20)->Item[6]
- several hundred nodes under Root->History->undo (20)->Item[7]
- 809199 nodes under Root->History->undo (20)->Item[8]
- 809199 nodes under Root->History->undo (20)->Item[9]
- 809199 nodes under Root->History->undo (20)->Item[10]
- 809199 nodes under Root->History->undo (20)->Item[11]
- 809199 nodes under Root->History->undo (20)->Item[12]
- several hundred nodes under Root->History->undo (20)->Item[13]
- several hundred nodes under Root->History->undo (20)->Item[14]
- several hundred nodes under Root->History->undo (20)->Item[15]
- 21021 nodes under Root->History->undo (20)->Item[16]
- several hundred nodes under Root->History->undo (20)->Item[17]
- 21316 nodes under Root->History->undo (20)->Item[18]
- 6 nodes under Root->History->undo (20)->Item[19]

You can see how, as history items get purged, the size of the history (and therefore the size of the .osp file) can change dramatically.

It looks like there are some operations which need to have the `ignore_history` flag set to True in the code so that those actions don't cause a huge history to be gathered.

I don't know, off the top of my head, how the history is built or in what order.  I'll have to research that more.

For now, @wolfnco , can you tell us what the "last 20" things you may have done were?  Did you drag and drop a bunch of files?  Move a bunch of clips on the timeline all at once with a multiple select?  Anything like that were you weren't just changing a single item.

Or, if you only changed single items, can you recall what items you may have changed?

**Comment by wolfnco** [comment]
Hello @N3WWN 
Sorry for the delay.
Thanks a lot for your analyze!

I'm trying a make a slidshow for a wedding. If I remember right, I made tests with adding about 30 pictures to timeline (using right click, add to timeline) with differents effects.

**Comment by N3WWN** [comment]
@wolfnco - Did you import them all at once via multiple selections in the file chooser or dragging and dropping all 30 files onto OpenShot?  Did you apply the effects one image at a time or did you select multiple images and then add the effects?

**Comment by wolfnco** [comment]
@N3WWN All at once, by dragging files to the project files window. And then select all files, right click, add to timeline, and choose effects in the new window.

**Comment by N3WWN** [comment]
@wolfnco - I just ran into this when working on a more-complex-than-my-usual-projects project... my .osp file was growing to over 45MB in size.  

In order to speed things up (and it sped things up IMMENSELY!) I went to `Edit`->`Preferences`->`Autosave`->`History Limit` and changed the value to `5`.  This meant that large batch operations would no longer result in **huge** recursively nested history that would be kept around for a long time.   I saved my project and relaunched OpenShot (to make sure the setting changed) and continued my work.

This took my .osp file down from 45MB to just under 300KB.  That's 1/150th the size!  The OpenShot interface became really responsive

@peanutbutterandcrackers - I think we have a definite bug to work on here.  This may be the cause of some of the performance issues remaining with the 2.4.2 release.  As a quick fix, we could adjust the value of `history-limit` down from `20` to something like `5` here : 
https://github.com/OpenShot/openshot-qt/blob/11cba82d0c77830fbf6495e26ca4eb87125e49ee/src/settings/_default.settings#L223
